src/r5py/r5/custom_cost_transport_network.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env python3
+
+"""Subclass for TransportNetwork, enables custom cost routing."""
+import com.conveyal.r5
+from r5py.r5.transport_network import TransportNetwork
+from r5py.util.custom_cost_conversions import (
+    convert_java_hashmap_to_python_dict,
+    convert_python_custom_costs_to_java_custom_costs,
+)
+from r5py.util.exceptions import CustomCostDataError
+from r5py.util.jvm import start_jvm
+
+__all__ = ["CustomCostTransportNetwork"]
+
+start_jvm()
+
+
+def r5_supports_custom_costs():
+    """
+    Check if the R5 java has the GP2 (Green Paths 2) patch i.e. supports custom costs routing.
+
+    Returns:
+    --------
+    bool: True if using GP2 R5, False otherwise.
+    """
+    try:
+        import com.conveyal.r5.rastercost.CustomCostField  # noqa: F401
+
+        # the import was successful thus using GP2 R5
+        return True
+    except ImportError:
+        # the import was unsuccessful
+        return False
+
+
+class CustomCostTransportNetwork(TransportNetwork):
+    """Inherit from TransportNetwork, adds custom cost data routing functionality."""
+
+    def __init__(self, osm_pbf, names, sensitivities, custom_cost_datas):
+        """
+        Initialise a transport network with custom costs.
+        Supports multiple custom costs. Must always have the same number of:
+        names, sensitivities, and custom_cost_datas.
+
+        Arguments
+        ---------
+        osm_pbf : str
+            file path of an OpenStreetMap extract in PBF format
+        names : List[str]
+            name(s) of the custom cost(s)
+        sensitivities : List[float] | List[int]
+            sensitivities of the custom cost(s)
+        custom_cost_datas : List[Dict[str, float]]
+            custom cost data(s) to be used in routing.
+            str key is osmid, float value is custom costs per edge (way).
+            multiple custom cost data can be provided as a list of python dicts,
+            when multiple custom cost datas are provided, all of those custom cost datas will be combined
+            for each edge (way) during r5 custom cost routing.
+        """
+        # crash if custom costs are NOT supported in the used version of R5
+        # either use TransportNetwork (without custom costs) or change to correct version of R5
+        if not r5_supports_custom_costs():
+            raise ImportError(
```

Comment:
Not sure whether `ImportError` is the right one here, could also be `RuntimeError`

Reply:
yeah true! Changing to RuntimeError